pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,774 @@
+	// pvc-protection, replicaset, replicationcontroller, resourcequota,
+	// root-ca-cert-publisher, route, service, serviceaccount, serviceaccount-token,
+	// statefulset, tokencleaner, ttl, ttl-after-finished
+	// Disabled-by-default controllers: bootstrapsigner, tokencleaner (default [*])
+	// Actual Supported controllers depend on the version of Kubernetes. See
+	// https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/
+	// for details.
+	//
+	// However, Karmada uses Kubernetes Native API definitions for federated resource template,
+	// so it doesn't need enable some resource related controllers like daemonset, deployment etc.
+	// On the other hand, Karmada leverages the capabilities of the Kubernetes controller to
+	// manage the lifecycle of the federated resource, so it needs to enable some controllers.
+	// For example, the `namespace` controller is used to manage the lifecycle of the namespace
+	// and the `garbagecollector` controller handles automatic clean-up of redundant items in
+	// your karmada.
+	//
+	// According to the user feedback and karmada requirements, the following controllers are
+	// enabled by default: namespace, garbagecollector, serviceaccount-token, ttl-after-finished,
+	// bootstrapsigner,csrapproving,csrcleaner,csrsigning. See
+	// https://karmada.io/docs/administrator/configuration/configure-controllers#kubernetes-controllers
+	//
+	// Others are disabled by default. If you want to enable or disable other controllers, you
+	// have to explicitly specify all the controllers that kube-controller-manager shoud enable
+	// at startup phase.
+	// +optional
+	Controllers []string `json:"controllers,omitempty"`
+
+	// ExtraArgs is an extra set of flags to pass to the kube-controller-manager component or
+	// override. A key in this map is the flag name as it appears on the command line except
+	// without leading dash(es).
+	//
+	// Note: This is a temporary solution to allow for the configuration of the
+	// kube-controller-manager component. In the future, we will provide a more structured way
+	// to configure the component. Once that is done, this field will be discouraged to be used.
+	// Incorrect settings on this feild maybe lead to the corresponding component in an unhealthy
+	// state. Before you do it, please confirm that you understand the risks of this configuration.
+	//
+	// For supported flags, please see
+	// https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/
+	// for details.
+	// +optional
+	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
+
+	// FeatureGates enabled by the user.
+	// More info: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/
+	// +optional
+	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+}
+
+// KarmadaControllerManager holds settings to the karmada-controller-manager component of the karmada.
+type KarmadaControllerManager struct {
```

Comment:
here, I have two not sure questions:
I think the name `KarmadaControllerManager ` is too long, can we simplify it to `ControllerManager`,  Other components are similar, such as: `KarmadaScheduler` =>  `Scheduler`,  `KarmadaAPIServer` => `APIServer`, but `KubeControllerManager`.

Reply:
I can tolerate the length compared to the clear meaning :).